pkg/controller/controller_binding_test.go

```diff
@@ -904,6 +977,124 @@ func TestReconcileBindingWithBrokerHTTPError(t *testing.T) {
+	assertNumberOfBrokerActions(t, brokerActions, 1)
+
+	events := getRecordedEvents(testController)
+	assertNumEvents(t, events, 1)
+
+	expectedEvent := api.EventTypeWarning + " " + errorBindCallReason + " " + "Error creating Binding \"test-binding/test-ns\" for Instance \"test-ns/test-instance\" of ServiceClass \"test-serviceclass\" at Broker \"test-broker\": fake creation failure"
+
+	if e, a := expectedEvent, events[0]; e != a {
+		t.Fatalf("Received unexpected event: %v", a)
+	}
+}
+
+func TestReconcileBindingWithBindingFailure(t *testing.T) {
+	fakeKubeClient, fakeCatalogClient, fakeBrokerClient, testController, sharedInformers := newTestController(t, fakeosb.FakeClientConfiguration{
+		BindReaction: &fakeosb.BindReaction{
+			Error: osb.HTTPStatusCodeError{
+				StatusCode:   http.StatusConflict,
+				ErrorMessage: strPtr("ServiceBindingExists"),
+				Description:  strPtr("Service binding with the same id, for the same service instance already exists."),
+			},
+		},
+	})
+
+	sharedInformers.Brokers().Informer().GetStore().Add(getTestBroker())
+	sharedInformers.ServiceClasses().Informer().GetStore().Add(getTestServiceClass())
+	sharedInformers.Instances().Informer().GetStore().Add(getTestInstanceWithStatus(v1alpha1.ConditionTrue))
+
+	binding := getTestBinding()
+
+	if err := testController.reconcileBinding(binding); err == nil {
+		t.Fatal("Binding creation should fail")
+	}
+
+	kubeActions := fakeKubeClient.Actions()
+	// one action: [{ActionImpl:{Namespace: Verb:get Resource:{Group: Version:v1 Resource:namespaces} Subresource:} Name:test-ns}]
+	assertNumberOfActions(t, kubeActions, 1)
+
+	actions := fakeCatalogClient.Actions()
+	// one action: [{ActionImpl:{Namespace:test-ns Verb:update Resource:{Group:servicecatalog.k8s.io Version:v1alpha1 Resource:bindings} Subresource:status} Object:0xc4200d02c0}]
+	assertNumberOfActions(t, actions, 1)
+
+	updatedObject := assertUpdateStatus(t, actions[0], binding)
+	assertBindingReadyFalse(t, updatedObject)
+	updatedBinding, ok := updatedObject.(*v1alpha1.Binding)
+	if !ok {
+		t.Fatal("Couldn't convert to v1alpha1.Binding")
+	}
+	if num := len(updatedBinding.Status.Conditions); num != 2 {
+		t.Fatalf("Expected two conditions, got %v", num)
+	}
+
+	brokerActions := fakeBrokerClient.Actions()
+	// actions: [{Type:Bind Request:0xc42013ed20}]
+	assertNumberOfBrokerActions(t, brokerActions, 1)
+
+	events := getRecordedEvents(testController)
+	assertNumEvents(t, events, 1)
+
+	expectedEvent := api.EventTypeWarning + " " + errorBindCallReason + " " + "Error creating Binding \"test-binding/test-ns\" for Instance \"test-ns/test-instance\" of ServiceClass \"test-serviceclass\" at Broker \"test-broker\", Status: 409; ErrorMessage: ServiceBindingExists; Description: Service binding with the same id, for the same service instance already exists.; ResponseError: <nil>"
+
+	if e, a := expectedEvent, events[0]; e != a {
+		t.Fatalf("Received unexpected event: %v", a)
+	}
+}
+
```

Comment:
There needs to be a test for `setBindingCondition` - you can crib from the one in #1017